Solution

olya-yevsieienko opened the pull request.

Olexandr6 commented on src/formatDate.js:
```diff
@@ -50,7 +50,29 @@
+  const separatorOldFormat = fromFormat[3];
+  const separatorNewFormat = toFormat[3];
+
+  const oldFormat = date.split(separatorOldFormat);
```
```suggestion
  const oldDateFormat = date.split(separatorOldFormat);
```

Olexandr6 commented on src/formatDate.js:
```diff
@@ -50,7 +50,29 @@
+  const newFormat = [];
+
+  for (let i = 0; i < oldFormat.length; i++) {
+    if (fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0) {
```
For such long condition better to create a separate variable to describe what you expect
```suggestion
		const isDifferentYearFormat = fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0;

    if (isDifferentYearFormat) {
```

Olexandr6 reviewed (changes requested):
Your solution is cool🔥 Let's work on some improvements

olya-yevsieienko commented:
Okey, I got it. Thanks!

вт, 14 июн. 2022 г. в 13:39, Olexandr Yatsykovych ***@***.***
>:

> ***@***.**** requested changes on this pull request.
>
> Your solution is cool🔥 Let's work on some improvements
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896656328>
> :
>
> > @@ -50,7 +50,29 @@
>
>   */
>
>
>
>  function formatDate(date, fromFormat, toFormat) {
>
> -  // write code here
>
> +  const separatorOldFormat = fromFormat[3];
>
> +  const separatorNewFormat = toFormat[3];
>
> +
>
> +  const oldFormat = date.split(separatorOldFormat);
>
>
> ⬇️ Suggested change
>
> -  const oldFormat = date.split(separatorOldFormat);
>
> +  const oldDateFormat = date.split(separatorOldFormat);
>
>
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896656483>
> :
>
> > @@ -50,7 +50,29 @@
>
>   */
>
>
>
>  function formatDate(date, fromFormat, toFormat) {
>
> -  // write code here
>
> +  const separatorOldFormat = fromFormat[3];
>
> +  const separatorNewFormat = toFormat[3];
>
> +
>
> +  const oldFormat = date.split(separatorOldFormat);
>
> +  const newFormat = [];
>
>
> ⬇️ Suggested change
>
> -  const newFormat = [];
>
> +  const newDateFormat = [];
>
>
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896657978>
> :
>
> > @@ -50,7 +50,29 @@
>
>   */
>
>
>
>  function formatDate(date, fromFormat, toFormat) {
>
> -  // write code here
>
> +  const separatorOldFormat = fromFormat[3];
>
> +  const separatorNewFormat = toFormat[3];
>
> +
>
> +  const oldFormat = date.split(separatorOldFormat);
>
> +  const newFormat = [];
>
> +
>
> +  for (let i = 0; i < oldFormat.length; i++) {
>
> +    if (fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0) {
>
>
> For such long condition better to create a separate variable to describe
> what you expect
> ⬇️ Suggested change
>
> -    if (fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0) {
>
> +		const isDifferentYearFormat = fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0;
>
> +
>
> +    if (isDifferentYearFormat) {
>
>
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896658978>
> :
>
> > @@ -50,7 +50,29 @@
>
>   */
>
>
>
>  function formatDate(date, fromFormat, toFormat) {
>
> -  // write code here
>
> +  const separatorOldFormat = fromFormat[3];
>
>
> ⬇️ Suggested change
>
> -  const separatorOldFormat = fromFormat[3];
>
> +  const oldDateFormatSeparator = fromFormat[3];
>
>
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896659379>
> :
>
> > @@ -50,7 +50,29 @@
>
>   */
>
>
>
>  function formatDate(date, fromFormat, toFormat) {
>
> -  // write code here
>
> +  const separatorOldFormat = fromFormat[3];
>
> +  const separatorNewFormat = toFormat[3];
>
>
> ⬇️ Suggested change
>
> -  const separatorNewFormat = toFormat[3];
>
> +  const newDateFormateJoiner = toFormat[3];
>
>
> ------------------------------
>
> In src/formatDate.js
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#discussion_r896655774>
> :
>
> > +    if (fromFormat[i].includes('Y') && toFormat.indexOf(fromFormat[i]) < 0) {
>
> +      if (fromFormat[i] === 'YYYY') {
>
>
> Yeah, better to avoid, but in this case, looks like impossible to avoid
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mate-academy/js_format_date_advanced/pull/620#pullrequestreview-1005628171>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO2L24OCXZHVHSVUULVCBR3VPBOOXANCNFSM5YUX4MTA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

Olexandr6 commented on src/formatDate.js:
```diff
@@ -50,7 +50,32 @@
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatoroldDateFormat = fromFormat[3];
```
```suggestion
  const separatorOldDateFormat = fromFormat[3];
```

Olexandr6 commented on src/formatDate.js:
```diff
@@ -50,7 +50,32 @@
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatoroldDateFormat = fromFormat[3];
+  const separatornewDateFormat = toFormat[3];
```
```suggestion
  const separatorNewDateFormat = toFormat[3];
```
